src/index.d.ts

```diff
@@ -0,0 +1,143 @@
+
+  /**
+   * Creates {@link Logger} with given options
+   * @param options the options (configs, etc ...) to initialize logger to be created
+   */
+  export function createLogger(options: any): Logger;
+
+  /**
+   * Creates an Express.js middleware to integrate Thundra
+   * @return the Thundra middleware
+   */
+  export function expressMW(): void;
+
+  /**
+   * Adds given log listener
+   * @param listener the log listener to be added
+   */
+  export function addLogListener(listener: any): void;
+
+  /**
+   * Gets the tracer
+   * @return {Tracer} the tracer
+   */
+  export function tracer(): ThundraTracer;
+
+  /**
+   * Loads and returns the user AWS Lambda handler
+   * @return the loaded user AWS Lambda handler
+   */
+  export function loadUserHandler(): any;
+
+  /**
+    * Thundra's logger implementation
+    */
+  export class Logger {
+    options: any;
+    loggerName: any;
+    logLevel: any;
+    levels: any;
+    constructor(options: {
+      loggerName: any;
+    });
+    /**
+     * Reports the given log
+     * @param level the log level
+     * @param args the log arguments
+     */
+    reportLog(level: any, args: any): void;
+    /**
+     * Logs in {@code TRACE} level
+     * @param args the log arguments
+     */
+    trace(...args: any[]): void;
+    /**
+     * Logs in {@code DEBUG} level
+     * @param args the log arguments
+     */
+    debug(...args: any[]): void;
+    /**
+     * Logs in {@code INFO} level
+     * @param args the log arguments
+     */
+    info(...args: any[]): void;
+    /**
+     * Logs in {@code WARN} level
+     * @param args the log arguments
+     */
+    warn(...args: any[]): void;
+    /**
+     * Logs in {@code ERROR} level
+     * @param args the log arguments
+     */
+    error(...args: any[]): void;
+    /**
+     * Logs in {@code FATAL} level
+     * @param args the log arguments
+     */
+    fatal(...args: any[]): void;
+    /**
+     * Logs directly by given arguments
+     * @param args the log arguments
+     */
+    log(...args: any[]): void;
+    /**
+     * Destroys plugin
+     */
+    destroy(): void;
+  }
+
+  /**
+   * Wraps the original AWS Lambda handler to hook into AWS Lambda invocation cycle
+   */
+  export interface WrappedFunction extends Function {
+    thundraWrapped?: boolean;
+  }
+
+  /**
+   * Thundra's {@link Tracer} implementation
+   */
+  export class ThundraTracer extends Tracer { }
+
+  /**
+   * Tracer is the entry-point between the instrumentation API and the tracing
+   * implementation.
+   *
+   * The default object acts as a no-op implementation.
+   *
+   * Note to implementators: derived classes can choose to directly implement the
+   * methods in the "OpenTracing API methods" section, or optionally the subset of
+   * underscore-prefixed methods to pick up the argument checking and handling
+   * automatically from the base class.
+   */
+  export class Tracer { }
```

Comment:
These classes, `ThundraTracer` and `Tracer`, need some dependencies before I include the definitions, but given that including those dependency files could increase our size and complexity, I've decided to leave it empty. 

See `tracer()` function.